### PR TITLE
Functionality Increment: Statistics piechart for room allocations

### PR DESCRIPTION
**Pull request by jetnew**
### What this does
This PR adds the functionality increment for the statistics piechart for room allocations. The piechart shows the ratio of allocated and unallocated rooms in ResiReg.

Part of #13.

### How to test

1. Launch the application.
2. Click the Statistics tab.
3. You should see a pie chart representing the number of allocated and unallocated rooms.
3. Run `allocate si/3 ri/3`.
4. Notice an update to the pie chart that now has 1 more allocated and 1 less unallocated room.

### Notes
image

### CODE REVIEW

File: src/main/java/seedu/resireg/ui/PiechartPanel.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        this.allocationList = allocationList;
+        this.roomList = roomList;
+
+        initData();
+        setChart();
+    }
+
+    private void initData() {
+        int allocatedCount = allocationList.size();
+        int unallocatedCount = roomList.size() - allocatedCount;
```
Comment (JingYenLoh) [inline src/main/java/seedu/resireg/ui/PiechartPanel.java:53]: might want to add an assertion here, that `unallocatedCount > 0`

File: src/main/java/seedu/resireg/ui/PiechartsTab.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package seedu.resireg.ui;
+
+/**
+ * Represents a tab containing a {@code piechartPanel}. The tab may contain other UI elements.
+ */
+interface PiechartsTab extends TabContainer {
+    void setPiechartPanel(PiechartPanel piechartPanel);
```
Comment (JingYenLoh) [inline src/main/java/seedu/resireg/ui/PiechartsTab.java:7]: What's the reason for this interface?

File: src/main/java/seedu/resireg/ui/PiechartPanel.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        data = FXCollections.observableArrayList();
+        data.add(new PieChart.Data("Allocated Rooms: " + allocatedCount, allocatedCount));
+        data.add(new PieChart.Data("Unallocated Rooms: " + unallocatedCount, unallocatedCount));
+    }
+
+    private void setChart() {
+        piechart.setData(data);
```
Comment (JingYenLoh) [inline src/main/java/seedu/resireg/ui/PiechartPanel.java:63]: Can consider logging here

**Review by JingYenLoh — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/resireg/ui/PiechartPanel.java (hunk 1 above)
Comment (jetnew) [reply src/main/java/seedu/resireg/ui/PiechartPanel.java:53]: I think that unallocatedCount can be 0.

File: src/main/java/seedu/resireg/ui/PiechartsTab.java (hunk 2 above)
Comment (jetnew) [reply src/main/java/seedu/resireg/ui/PiechartsTab.java:7]: The interface is for future-proofing if we ever want to have a list of piecharts.

File: src/main/java/seedu/resireg/ui/PiechartPanel.java (hunk 1 above)
Comment (JingYenLoh) [reply src/main/java/seedu/resireg/ui/PiechartPanel.java:53]: Good point!

File: src/main/java/seedu/resireg/ui/PiechartPanel.java (hunk 3 above)
Comment (jetnew) [reply src/main/java/seedu/resireg/ui/PiechartPanel.java:63]: Added logging for setting chart data

File: src/main/java/seedu/resireg/ui/PiechartPanel.java (hunk 1 above)
Comment (jetnew) [reply src/main/java/seedu/resireg/ui/PiechartPanel.java:53]: Added assertion for `unallocatedCount >= 0`.

**Review by JingYenLoh — APPROVED** [review]

**Pull request merged**